### ISSUE TITLE
Performance on Volta GPUs

### ISSUE THREAD

**Issue by aaronzira** [SeanNaren/deepspeech.pytorch#167]
I've tried a few runs on P3.2x and P3.16x (Volta) EC2 spot instances since they became available this week, and am observing unexpectedly slow data load times, as well as uniformly 0.0000 loss values per batch on TED data. Without the --cuda flag, load times are much slower, but the loss values are reasonable. Everything is using most recent CUDA 9.0, cuDNN 7.0.3, pytorch, and deepspeech.pytorch master as of around 34 hours ago, on Ubuntu 16.04.

- Without CUDA
> Epoch: [1][1/4649]      Time 96.611 (96.611)    Data 0.353 (0.353)      Loss 477.5085 (477.5085)
> Epoch: [1][2/4649]      Time 275.491 (186.051)  Data 0.060 (0.206)      Loss 576.2980 (526.9033)

- With CUDA (load times do eventually stabilize at the 1-2s range)
> Epoch: [1][1/4649]      Time 17.188 (17.188)    Data 3.205 (3.205)      Loss 0.0000 (0.0000)
> Epoch: [1][2/4649]      Time 1.536 (9.362)      Data 0.006 (1.606)      Loss 0.0000 (0.0000)
> Epoch: [1][3/4649]      Time 1.199 (6.641)      Data 0.015 (1.075)      Loss 0.0000 (0.0000)
> Epoch: [1][4/4649]      Time 0.710 (5.158)      Data 0.009 (0.809)      Loss 0.0000 (0.0000)

Using the same data on a 4xGTX 1080Ti (Pascal) setup, load times per batch start in the neighborhood of 0.115s.

Could this have something to do with CUDA 9.0 or the fact that Volta is different? Based on what I've read with other comparisons, I would expect performance to be significantly better with the new architecture.

**Comment by jinserk** [comment]
Did you change hyper parameters such as lr or batch size? I'm running cuda 9.0 with gtx 1070 and sometimes I've got the loss of 0 or even NaN with large lr.

**Comment by SeanNaren** [comment]
This might be a warp-ctc compatibility issue due to the architecture numbers, will look into it!

**Comment by SeanNaren** [comment]
Repull the warp-ctc pytorch bindings, and re-install. Should be fixed

**Comment by SeanNaren** [comment]
I've managed to get training working with the new warp-ctc bindings, so I'll close this for now

**Comment by aaronzira** [comment]
Awesome -- looks great! Thank you!!

**Comment by geniass** [comment]
This doesn't seem to be fixed with the latest bindings.

I'm running Ubuntu 16.04 on a p3.2xlarge with CUDA 9. Even after pulling the latest version of everything (warp-ctc, audio, ctcdecode and this repo as of now) I still get nan loss after a few batches on the Librispeech data.

I also tried rebuilding pytorch from source which didn't help.
